tests/Unit/Evolution/DgSubcell/Test_Projection.cpp

```diff
@@ -0,0 +1,154 @@
+    CHECK_ITERABLE_APPROX(
+        cell_average_times_volume,
+        TestHelpers::evolution::dg::subcell::cell_values(
+            dg_mesh.extents(0) - 2, logical_coordinates(subcell_mesh)));
+
+    // Test projection of a Variables using the return-by-value function
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>> vars(
+        dg_mesh.number_of_grid_points());
+    get(get<Tags::Scalar>(vars)) =
+        TestHelpers::evolution::dg::subcell::cell_values(dg_mesh.extents(0) - 2,
+                                                         logical_coords);
+    for (size_t d = 0; d < Dim; ++d) {
+      get<Tags::Vector<Dim>>(vars).get(d) =
+          (d + 2.0) * TestHelpers::evolution::dg::subcell::cell_values(
+                          dg_mesh.extents(0) - 2, logical_coords);
+    }
+    Variables<tmpl::list<Tags::Scalar, Tags::Vector<Dim>>>
+        cell_average_times_volume_vars = evolution::dg::subcell::fd::project(
+            vars, dg_mesh, subcell_mesh.extents());
+
+    const auto check_impl = [&dg_mesh,
+                             &subcell_mesh](const auto& subcell_vars) {
+      CHECK_ITERABLE_APPROX(
+          get(get<tmpl::front<
+                  typename std::decay_t<decltype(subcell_vars)>::tags_list>>(
+              subcell_vars)),
+          TestHelpers::evolution::dg::subcell::cell_values(
+              dg_mesh.extents(0) - 2, logical_coordinates(subcell_mesh)));
+      for (size_t d = 0; d < Dim; ++d) {
+        CHECK_ITERABLE_APPROX(
+            get<tmpl::back<
+                typename std::decay_t<decltype(subcell_vars)>::tags_list>>(
+                subcell_vars)
+                .get(d),
+            (d + 2.0) *
+                TestHelpers::evolution::dg::subcell::cell_values(
+                    dg_mesh.extents(0) - 2, logical_coordinates(subcell_mesh)));
+      }
+    };
+    check_impl(cell_average_times_volume_vars);
+
+    Variables<
+        tmpl::list<Tags::Prefix<Tags::Scalar>, Tags::Prefix<Tags::Vector<Dim>>>>
+        prefixed_cell_average_times_volume_vars{
+            subcell_mesh.number_of_grid_points()};
+    evolution::dg::subcell::fd::project(
+        make_not_null(&prefixed_cell_average_times_volume_vars), vars, dg_mesh,
+        subcell_mesh.extents());
+    check_impl(prefixed_cell_average_times_volume_vars);
+
+    Variables<
+        tmpl::list<Tags::Prefix<Tags::Scalar>, Tags::Prefix<Tags::Vector<Dim>>>>
+        prefixed_vars(dg_mesh.number_of_grid_points());
+    prefixed_vars = vars;
+    cell_average_times_volume_vars.initialize(
+        0.0, subcell_mesh.number_of_grid_points());
+    evolution::dg::subcell::fd::project(
+        make_not_null(&cell_average_times_volume_vars), prefixed_vars, dg_mesh,
+        subcell_mesh.extents());
+    check_impl(cell_average_times_volume_vars);
+
+    prefixed_cell_average_times_volume_vars.initialize(
+        0.0, subcell_mesh.number_of_grid_points());
+    evolution::dg::subcell::fd::project(
```

Comment:
I'm confused about what these last three projections are checking. Can you add short comments as you did for the DataVector and the first Vars case?